packages/bierzo-wallet/src/store/tokens/actions.ts

```diff
@@ -1,18 +1,45 @@
+import { BlockchainConnection } from '@iov/bcp';
+import { BnsConnection } from '@iov/bns';
 import { EthereumConnection } from '@iov/ethereum';
+import { LiskConnection } from '@iov/lisk';
+import { ChainSpec, loadConfig } from '../../utils/config';
 import { AddTickerActionType, BwToken } from './reducer';
 
-export async function getTokens(): Promise<{ [key: string]: BwToken }> {
+async function createConnectionFor(spec: ChainSpec): Promise<BlockchainConnection | null> {
+  const url = spec.bootstrapNodes[0];
+  if (spec.codecType === 'eth') {
+    return EthereumConnection.establish(url, {});
+  }
+
+  if (spec.codecType === 'bns') {
+    return BnsConnection.establish(url);
+  }
+
+  if (spec.codecType === 'lsk') {
+    return LiskConnection.establish(url);
+  }
+
+  return null;
+}
+
+export async function getTokens(): Promise<{ [ticker: string]: BwToken }> {
+  const config = await loadConfig();
   const tokens: { [ticker: string]: BwToken } = {};
+  const chains = config.chains;
+
+  for (const chain of chains) {
+    const connection = await createConnectionFor(chain.chainSpec);
```

Comment:
The `connection` created here needs to be disconnected by calling `connection.disconnect()` when done. It would be better to keep the connects alive and reuse them later